### PR TITLE
respect SKIMAGE_TEST_STRICT_WARNINGS_GLOBAL setting in tests.yml

### PR DESCRIPTION
**Pull request by grlee77**
## Description

Some time ago we added a CI case that was supposed to raise errors on unexpected warnings. As noticed in #6116, this has not been running and there are a few new warnings showing up in the test suite.

I am first running CI with just a commit to get the warnings test working. This should fail as-is. Then I will add follow-up commits to fix the warnings causing the failure.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by grlee77** [comment]
Not sure why the --pre case that has SKIMAGE_TEST_STRICT_WARNINGS_GLOBAL=1 is showing up as passing. There are 29 errors showing up in the log here: https://github.com/scikit-image/scikit-image/runs/4473613135?check_suite_focus=true#step:5:15242

**Comment by grlee77** [comment]
@hmaarrfk, do you have any idea why the test runs corresponding to 0a12937 above show as successful even though in some cases pytest failed during test collection?

I think the warnings have been fixed at this point (aside from ones we were already skipping previously).

**Comment by grlee77** [comment]
Weird, I put an intentional error in a Gaussian filter test case and still all tests pass on GitHub Actions! It is at least failing on the other CI, though.

**Comment by hmaarrfk** [comment]
Sorry I'm not sure how things fail during collection.

**Comment by grlee77** [comment]
Well the list commit is a failure not during collection but directly in the test and it still passed

**Comment by hmaarrfk** [comment]
Funny how this kinda goes round and round in a circle:

1. New CI makes people excited.
2. People are tired of the old build scripts.
3. People simplify the build script into the CI configuration file.
4. Some strange edge case in the CI configuration file makes us go back to build scripts.

**Comment by grlee77** [comment]
@scikit-image/core: It is high priority to get this one in and then backport to v0.19.x so that all tests are in a working state. Please take a look!

Now that everything is passing I rebased to clean things up and condense the changes related to GitHub Actions into the first commit.  

The other commits are each very small and can be reviewed individually. They address several different warnings/errors that are needed for all tests to pass. There are no changes in behavior to user-facing functions. The changes here are purely to avoid warnings (or in one case using a stable seed for an example to avoid a potential stochastic test failure).

File: skimage/_shared/tests/test_utils.py [hunk 1]
```diff
@@ -152,24 +144,24 @@ def bar(arg0, new_arg1=1, arg2=None):
 
     # Assert that nothing happens when the function is called with the
     # new API
-    with pytest.warns(None) as record:
```
Comment (hmaarrfk) [inline skimage/_shared/tests/test_utils.py:155]: Don't we still need this pytest warning assertion?

Comment (grlee77) [reply skimage/_shared/tests/test_utils.py:155]: Well, it should fail for the CI job with strict warnings enabled. To keep it equivalent to before, let me change it to use `warnings.catch_warnings` so we can still demonstrate that there are none raised

Comment (grlee77) [reply skimage/_shared/tests/test_utils.py:155]: see a49e9b5 for updated code that still checks the recorded warnings to make sure there are no

**Review by hmaarrfk — APPROVED** [review]
I added back one check. Feel free to rebase if you want.

**Comment by grlee77** [comment]
Thanks @hmaarrfk, I had missed that one!

**Comment by grlee77** [comment]
The single macos-cp3.9 failure here is stochastic and would be resolved by #6114

**Pull request merged**